QModManager/Utility/MainMenuMessages.cs

```diff
@@ -0,0 +1,169 @@
+﻿namespace QModManager.Utility
+{
+    using System.Reflection;
+    using System.Reflection.Emit;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    using Harmony;
+    using UnityEngine;
+    using UnityEngine.SceneManagement;
+
+    /// <summary>
+    /// Allows to add critical messages to the main menu.
+    /// Messages will stay in the main menu and on loading screen.
+    /// </summary>
+    public static class MainMenuMessages
```

Comment:
This is a `public` feature for use by other mods.  
Move it into the `API` folder & namespace (along with all the other public classes).

I would also recommend making this functionality accessible through the [IQModServices](https://github.com/SubnauticaModding/QModManager/blob/master/QModManager/API/IQModServices.cs) interface.  
Just add the necessary methods to the interface and implement them in [QModServices](https://github.com/SubnauticaModding/QModManager/blob/master/QModManager/API/QModServices.cs) class.

Reply:
> "Move it into the API folder" vs "No need to move"

So, to move or not to move? :)
I think, since `MainMenuMessages` have only one public method I can make this class internal and keep it in Utility namespace and add public method to QModServices as a pass-through. 

Am I understand you correctly?

Reply:
Yeah, sorry. I made things confusing there.  
I said two things cause I was implicitly presenting two options.

The main two things to end up with are:
1. Any public `class` should only exist in (or be accessible through) the `API` namespace.
2. QModServices should be able to expose any public functionality, even if that functionality is implemented somewhere else, just to make discoverability easier.

**Option 1**
Yes, you can absolutely make `MainMenuMessages` be `internal` and have `QModServices` call its methods as a pass-through.

**Option 2**
If you would prefer to keep the `MainMenuMessages` class as `public` and have it usable directly (in addition to `QModServices`), then _that_ is when I would say it should be moved into the `API` namespace.

Reply:
Ok, I'll go with option 1 then